packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -931,14 +931,14 @@ test.serial('writeGranulesFromMessage() writes a granule and marks as failed if
 
   const dynamoGranule = await granuleModel.get({ granuleId });
   t.is(dynamoGranule.status, 'failed');
-  t.deepEqual(dynamoGranule.error.Error, 'Failed writing files to PostgreSQL.');
+  t.deepEqual(JSON.parse(dynamoGranule.error.Errors)[0].Error, 'Failed writing files to PostgreSQL.');
```

Comment:
```suggestion
  t.deepEqual(JSON.parse(dynamoGranule.error.Errors), [{ Error: 'Failed writing files to PostgreSQL.' }]);
```

I think we should compare against the entire `Errors` array so that we can be sure that only the errors we expect are in the array and nothing else. Same for the rest of the tests in the file

Separately, I think we need a new unit test to simulate a scenario where there is a workflow error and an error writing files to the database to verify that both errors appear in `Errors`

Reply:
[`65786c0` (#2567)](https://github.com/nasa/cumulus/pull/2567/commits/65786c0d912d5deb00162f4b6982906b60c341d8#diff-e4b95ee0e3b92433a8e670895c1e8da32d083c3ca64e31761d4bb353cbb791a7R933)

Reply:
[`36464f8` (#2567)](https://github.com/nasa/cumulus/pull/2567/commits/36464f849d127e5b77fcb23e44ea6f56d907f328) I added a unit test for the second case you mentioned. Let me know what you think about the use of `truthy` on `Cause`